# Урок 3

In [138]:
import numpy as np  # was used to check manually counted answers
from math import factorial

In [139]:
def comb(n, k):
    return factorial(n) / (factorial(k) * factorial(n - k))

### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [140]:
print(math.sqrt(25))

5.0


In [141]:
a = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [142]:
mean_a = sum(a) / len(a)
print(f'mean - {mean_a}')

mean - 65.3


In [143]:
a_sorted = a.copy()
a_sorted.sort()

a_temp = [el - mean_a for el in a_sorted]

a_temp2 = [el ** 2 for el in a_temp]

var_a = sum(a_temp2) / len(a)

print(f'variance - {var_a}')

variance - 950.11


In [144]:
print(f'standard diviation - {math.sqrt(var_a)}')

standard diviation - 30.823854398825596


In [145]:
var_a_not_displaced = sum(a_temp2) / (len(a) - 1)
print(f'not displaced variance - {var_a_not_displaced}')

not displaced variance - 1000.1157894736842


Среднее арифметическое - 65.3<br>
Среднее квадратичное отклонение - 30.82<br>
Дисперсия - 950.11<br>
Несмещенная дисперсия - 1000.12

### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [146]:
p1 =                            (5/8 * 4/7) * (comb(5, 1) * comb(7, 3) / comb(12, 4))
p2 = (comb(5, 1) * comb(3, 1) / comb(8, 2)) * (comb(5, 2) * comb(7, 2) / comb(12, 4))
p3 =                            (3/8 * 2/7) * (comb(5, 3) * comb(7, 1) / comb(12, 4))

p = p1 + p2 + p3

print(p1, p2, p3, p)

0.12626262626262627 0.22727272727272727 0.01515151515151515 0.3686868686868687


**Вероятность что 3 из 6 вытянутых мячей белые - 36.9%**

<img src='./img_l3/2.jpg' style='width: 600px;'>

### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [147]:
p_a = 1/3 * 9/10 + 1/3 * 8/10 + 1/3 * 6/10

p_ab1 = 1/3 * 9/10 / p_a
p_ab2 = 1/3 * 8/10 / p_a
p_ab3 = 1/3 * 6/10 / p_a

print(p_a, p_ab1, p_ab2, p_ab3)

0.7666666666666666 0.391304347826087 0.3478260869565218 0.26086956521739135


Вероятность, что попавший выстрел произведен:<br>
а) первым спортсменом - **39.1%**<br>
б) вторым спортсменом - **34.8%**<br>
в) третьим спортсменом - **26.1%**

<img src='./img_l3/3.jpg' style='width: 600px;'>

### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [148]:
p_a = 1/4 * 8/10 + 1/4 * 7/10 + 2/4 * 9/10

p_ab1 = 1/4 * 8/10 / p_a
p_ab2 = 1/4 * 7/10 / p_a
p_ab3 = 1/2 * 9/10 / p_a

print(p_a, p_ab1, p_ab2, p_ab3)

0.825 0.24242424242424246 0.21212121212121213 0.5454545454545455


Вероятность того, что сдавший первую сессию студент:<br>
а) на факультете А - **24.2%**<br>
б) на факультете В - **21.2%**<br>
в) на факультете С - **54.5%**

<img src='./img_l3/4.jpg' style='width: 600px;'>

### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

### а)

In [149]:
p = 0.1 * 0.2 * 0.25

print(f'Вероятность что все детали сломаются:\n{p}')

Вероятность что все детали сломаются:
0.005000000000000001


### б)

In [150]:
p1 = 1/10 * 2/10 * 3/4
p2 = 1/10 * 8/10 * 1/4
p3 = 9/10 * 2/10 * 1/4

p = p1 + p2 + p3

print(f'Вероятность что две детали сломаются:\n{p}')

Вероятность что две детали сломаются:
0.08


### в)

In [151]:
# тут р - это вероятность того что все детали НЕ сломаются
p = 0.9 * 0.8 * 0.75

print(f'Вероятность что ни одна деталь не сломается:\n{p}')

Вероятность что ни одна деталь не сломается:
0.54


### г)

In [152]:
p11 = 1/10 * 8/10 * 3/4
p12 = 9/10 * 2/10 * 3/4
p13 = 9/10 * 8/10 * 1/4

p1 = p11 + p12 + p13
p2 = 0.08

p = p1 + p2

print(f'Вероятность что одна-две детали сломаются:\n{p}')

Вероятность что одна-две детали сломаются:
0.455


In [153]:
print(f'Проверка:\nот единицы отнимает вероятность, что все детали уцелеют,\n\
сравниваем суммой вероятности того, что все детали сломаются\n\
и вероятностью что 1-2 детали сломаются\n\
1 - в) = а) + г)\n\
{1 - 0.54} = {0.005 + 0.455}')

Проверка:
от единицы отнимает вероятность, что все детали уцелеют,
сравниваем суммой вероятности того, что все детали сломаются
и вероятностью что 1-2 детали сломаются
1 - в) = а) + г)
0.45999999999999996 = 0.46


а) Вероятность что все детали сломаются - **0.005**<br>
б) вероятность что две детали сломаются - **0.08**<br><br>
_вероятность что одна деталь сломается - **0.375**_<br><br>
г) вероятность что одна-две детали сломаются - **0.455**<br><br>
в) вероятность того, что ни одна не сломается - **0.54** (1 - 0.455 + 0.005)<br>

<img src='./img_l3/5.jpg' style='width: 100%;'>